### PR TITLE
Add domain option for resample.

### PR DESCRIPTION
**Pull request by luigifcruz**
The original `scipy.signal.resample` function converts the input time-domain signal into frequency-domain to perform the resampling. This pull request proposes to add an option `domain` to pass a signal already in the frequency-domain. This would reduce the computational load of workloads that require several resamples of the same input signal. Here, the `domain` option is passed to the function with `time` for time-domain (Default) or `freq` for frequency-domain.

### CODE REVIEW

**Review by miladsade96 — CHANGES_REQUESTED** [review]
@luigifreitas Please add tests related to your feature into **/scipy/signal/tests/test_signaltools.py**

File: scipy/signal/signaltools.py [hunk 1]
```diff
@@ -2803,6 +2803,13 @@ def resample(x, num, t=None, axis=0, window=None):
     window : array_like, callable, string, float, or tuple, optional
         Specifies the window applied to the signal in the Fourier
         domain.  See below for details.
+    domain : string, optional
+
```
Comment (EverLookNeverSee) [inline scipy/signal/signaltools.py:2807]:
```suggestion
```

Comment (EverLookNeverSee) [reply scipy/signal/signaltools.py:2807]: remove this blank line

File: scipy/signal/signaltools.py [hunk 2]
```diff
@@ -2803,6 +2803,13 @@ def resample(x, num, t=None, axis=0, window=None):
...
+        A string indicating the domain of the input `x`:
+        ``time``
+           Consider the input `x` as time-domain. (Default)
+        ``freq``
+           Consider the input `x` as frequency-domain.
```
Comment (EverLookNeverSee) [inline scipy/signal/signaltools.py:2812]:
```suggestion
        ``time`` Consider the input `x` as time-domain (Default),
        ``freq`` Consider the input `x` as frequency-domain.
```

**Review by miladsade96 — CHANGES_REQUESTED** [review]

**Comment by e-q** [comment]
I see the logic in wanting to spare redundant DFT calculations, but the keyword argument approach seems a little awkward to me. One alternative could be to factor out some of the inner logic of `resample` into a `resample_dft` function that takes in the user's DFT and returns the `Y` currently calculated in `resample`. Then, `resample` could call that within itself and handle the fft/ifft calculations.

**Review by ilayn — CHANGES_REQUESTED** [review]

File: scipy/signal/signaltools.py [hunk 3]
```diff
@@ -2867,11 +2874,16 @@ def resample(x, num, t=None, axis=0, window=None):
...
+    if domain == 'time':
+        # Forward transform
+        if real_input:
+            X = sp_fft.rfft(x, axis=axis)
+        else:  # Full complex FFT
+            X = sp_fft.fft(x, axis=axis)
+    elif domain == 'freq':
+        X = x
+    else:
+        raise NotImplementedError("domain should be 'time' or 'freq'")
```
[the remark this answers is not in the document]
Comment (luigifreitas) [reply scipy/signal/signaltools.py]: Domain ValueError validation added as requested.

File: scipy/signal/tests/test_signaltools.py [hunk 4]
```diff
@@ -16,6 +16,7 @@
 from numpy import array, arange
 import numpy as np
 
+from scipy import fftpack
```
Comment (ilayn) [inline scipy/signal/tests/test_signaltools.py:19]: Just import `scipy.fft`

File: scipy/signal/signaltools.py [hunk 5]
```diff
@@ -2862,17 +2866,25 @@ def resample(x, num, t=None, axis=0, window=None):
...
-        X = sp_fft.rfft(x, axis=axis)
-    else:  # Full complex FFT
-        X = sp_fft.fft(x, axis=axis)
+    if domain == 'time':
+        # Forward transform
+        if real_input:
+            X = sp_fft.rfft(x, axis=axis)
+        else:  # Full complex FFT
+            X = sp_fft.fft(x, axis=axis)
+    elif domain == 'freq':
```
Comment (ilayn) [inline scipy/signal/signaltools.py:2886]: Now it is just `else`. You can write 

```python
    else:  # domain == 'freq'
```
Note double whitespace before `#` for PEP8

**Review by ilayn — CHANGES_REQUESTED** [review]
Only minor comments

**Comment by ilayn** [comment]
Thank you @luigifreitas for your time!

**Pull request merged**